AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -749,22 +762,27 @@ - (BOOL)shouldBatchFetch
   }
 }
 
-- (void)handleBatchFetchScrollingToOffset:(CGPoint)targetOffset
+- (void)_handleBatchFetchScrollingToOffset:(CGPoint)targetOffset
 {
   ASDisplayNodeAssert(_batchContext != nil, @"Batch context should exist");
   
-  if (![self shouldBatchFetch]) {
+  if (![self _shouldBatchFetch]) {
     return;
   }
   
   if (ASDisplayShouldFetchBatchForContext(_batchContext, [self scrollDirection], self.bounds, self.contentSize, targetOffset, _leadingScreensForBatching)) {
```

Comment:
This call point is very similar to _checkForBatchFetching.  Let's unify them, maybe remove _checkForBatchFetching and this method and replace with _batchFetchIfNeededForContentOffset:.  You could just pass as the argument self.contentOffset when not using targetOffset.